### PR TITLE
When to Use a Thread-Safe Collection

### PR DESCRIPTION
**Pull request by stevehoag**
Fixed <> formatting.

### CODE REVIEW

File: docs/essentials/collections/threadsafe/whentoUseaThread-SafeCollection.md [hunk 1]
```diff
@@ -1 +1,55 @@
...
+
+In general, use a [System.Collections.Concurrent.ConcurrentDictionary&lt;TKey, TValue&gt;](http://dotnet.github.io/api/System.Collections.Concurrent.ConcurrentDictionary%602.html) in any scenario where you are adding and updating keys or values concurrently from multiple threads. In scenarios that involve frequent updates and relatively few reads, the **ConcurrentDictionary&lt;TKey, TValue&gt;** generally offers modest benefits. In scenarios that involve many reads and many updates, the **ConcurrentDictionary&lt;TKey, TValue&gt;** generally is significantly faster on computers that have any number of cores.
+
+In scenarios that involve frequent updates, you can increase the degree of concurrency in the **ConcurrentDictionary&lt;TKey, TValue&gt;** and then measure to see whether performance increases on computers that have more cores. If you change the concurrency level, avoid global operations as much as possible.
+
+If you are only reading key or values, the [Dictionary&lt;TKey, TValue&gt;](http://dotnet.github.io/api/System.Collections.Generic.Dictionary%602.html) is faster because no synchronization is required if the dictionary is not being modified by any threads.
```
Comment (cartermp) [inline docs/essentials/collections/threadsafe/whentoUseaThread-SafeCollection.md]: Here `Dictionary<TKey, TValue>` doesn't have the fully-qualified namespace.  That should be added for consistency.

Also I think another discussion item for this week's meeting is using fully-qualified namespaces by default vs only using them when there's ambiguity.

**Comment by cartermp** [comment]
LGTM modulo feedback.

File: docs/essentials/collections/threadsafe/whentoUseaThread-SafeCollection.md (hunk 1 above)
Comment (mairaw) [inline docs/essentials/collections/threadsafe/whentoUseaThread-SafeCollection.md]: On MSDN, we don't usually add the namespace unless it's needed for disambiguation or understanding. @stevehoag just copied how it was done in the original article but I actually dislike the fully-qualified name.

Comment (stevehoag) [inline docs/essentials/collections/threadsafe/whentoUseaThread-SafeCollection.md]: I like the idea of using the fully qualified name on first mention, as it reinforces memory of what namespace it's in, especially in the case where we have several similarly named namespaces. I've also been following the convention of linking on first mention, then bolding further mentions.

We should discuss at our next meeting and establish style guidelines for consistency.

Comment (cartermp) [inline docs/essentials/collections/threadsafe/whentoUseaThread-SafeCollection.md]: Yes, we should discuss what we think is best.

**Pull request merged**